src/Features/Core/Portable/QuickInfo/QuickInfoOptions.cs

```diff
@@ -4,18 +4,17 @@
 
 using System.Runtime.Serialization;
 
-namespace Microsoft.CodeAnalysis.QuickInfo
+namespace Microsoft.CodeAnalysis.QuickInfo;
+
+[DataContract]
+internal readonly record struct QuickInfoOptions
 {
-    [DataContract]
-    internal readonly record struct QuickInfoOptions(
-        [property: DataMember(Order = 0)] bool ShowRemarksInQuickInfo = true,
-        [property: DataMember(Order = 1)] bool IncludeNavigationHintsInQuickInfo = true)
```

Comment:
❔ Why do we not want to keep this pattern?

Reply:
Primary constructors do not work well with inheritance and non-constant default values. In this particular case it's not an issue (yet) but other options run into those problems. If a property is added that requires non-constant default value it would also become an issue. It's better to use init-only properties and object initializers consistently for all option types.